sdk/tests/psp22.test.ts

```diff
@@ -0,0 +1,75 @@
+import { ApiPromise, Keyring } from '@polkadot/api'
+import { IKeyringPair } from '@polkadot/types/types/interfaces'
+import { expect } from 'chai'
+import { Network } from '../src/network'
+import { PSP22 } from '../src/psp22'
+import { getDeploymentData, initPolkadotApi } from '../src/utils'
+
+describe('psp22', function () {
+  const init = async (): Promise<{
+    api: ApiPromise
+    account: IKeyringPair
+    testAccount: IKeyringPair
+  }> => {
+    const api = await initPolkadotApi(Network.Local)
+
+    const keyring = new Keyring({ type: 'sr25519' })
+    const account = await keyring.addFromUri('//Alice')
+    const testAccount = await keyring.addFromUri('//Bob')
+
+    return { api, account, testAccount }
+  }
+
+  it('deploys', async () => {
+    const { api, account } = await init()
+
+    const tokenData = await getDeploymentData('psp22')
+    const token = new PSP22(api, Network.Local)
+
+    const name = api.createType('Option<String>', 'Coin')
+    const symbol = api.createType('Option<String>', 'COIN')
+
+    const tokenDeploy = await token.deploy(
+      account,
+      tokenData.abi,
+      tokenData.wasm,
+      1000,
+      name,
+      symbol,
+      12
+    )
+    await token.load(tokenDeploy.address, tokenData.abi)
+  })
+
+  it('works', async () => {
```

Comment:
If any action can be broken down into smaller steps, I think it's worth doing so.  It can be divided into get metadata, mint, and transfer